libs/ts/contracts/tasks/deployment-utils/init-chain.ts

```diff
@@ -1,101 +1,132 @@
-import type { Network, Signer } from 'ethers';
-import { Wallet, JsonRpcProvider } from 'ethers';
+import { Schema as S } from 'effect';
 
+import type { ethers as _ethers } from 'ethers';
+import { Wallet, JsonRpcProvider } from 'ethers';
 import type { HardhatEthersHelpers } from '@nomicfoundation/hardhat-ethers/types';
 
 import {
-  getRpcUrl,
-  kebabToScreamingSnakeCase,
+  withTimeout,
   parseEthereumAddress,
-  getOptionalEnvString,
-  getEnvString,
   NetworkName,
+  fromCommaSeparatedString,
+  ethereumAddress,
+  asVarSchema,
+  hexDataString,
+  networkName,
+  parseHexDataString,
 } from '@blocksense/base-utils';
 
-import { awaitTimeout } from '../utils';
-import { NetworkConfig } from '../types';
+import {
+  DeploymentEnvSchema,
+  parseDeploymentEnvConfig,
+  validateAndPrintDeploymentEnvConfig,
+} from '@blocksense/base-utils/evm/functions';
+
+import type { NetworkConfig } from '../types';
+
+const sharedPerNetworkKind = {
+  deployerAddressIsLedger: asVarSchema(S.BooleanFromString),
+  deployerAddress: ethereumAddress,
+  deployerPrivateKey: hexDataString,
+
+  adfsUpgradeableProxySalt: asVarSchema(hexDataString),
+
+  adminMultisigThreshold: S.NumberFromString,
+  adminMultisigOwners: fromCommaSeparatedString(ethereumAddress),
+
+  sequencerAddress: ethereumAddress,
+
+  reporterMultisigEnable: asVarSchema(S.BooleanFromString),
+  reporterMultisigThreshold: S.NumberFromString,
+  reporterMultisigSigners: asVarSchema(
+    fromCommaSeparatedString(ethereumAddress),
+  ),
+
+  isSafeOriginalDeployment: asVarSchema(S.BooleanFromString),
+};
+
+const envSchema = {
+  global: {
+    NETWORKS: fromCommaSeparatedString(networkName),
+  },
+
+  perNetworkKind: sharedPerNetworkKind,
+
+  perNetworkName: {
+    rpcUrl: S.URL,
+    feedIds: S.Union(S.Literal('all'), fromCommaSeparatedString(S.BigInt)),
+
+    ...sharedPerNetworkKind,
+  },
+} satisfies DeploymentEnvSchema;
 
 export async function initChain(
-  ethers: HardhatEthersHelpers,
+  ethers: typeof _ethers & HardhatEthersHelpers,
```

Comment:
Maybe `ethers_` is more suitable - I've seen it in many languages when the name is the same as another variable.

Reply:
Why not keep the `import type {...}` and just update the line below with ethers functions and classes?

```suggestion
import type { ethers as EthersType } from 'ethers';
import { Wallet, JsonRpcProvider, id } from 'ethers';
```